plugins/inputs/snmp/snmp.go

```diff
@@ -560,9 +565,8 @@ func (s *Snmp) getConnection(idx int) (snmpConnection, error) {
 	if err != nil {
 		return nil, err
 	}
-	gs.SetAgent(agent)
-	if err != nil {
-		return nil, err
+	if err := gs.SetAgent(agent); err != nil {
+		return nil, fmt.Errorf("setting gosnmpwrapper agent: %w", err)
 	}
```

Comment:
I think the idea was to behave it similar like the lines above, so it should have been `err = gs.SetAgent(agent)`